synapse/handlers/room_summary.py

```diff
@@ -1096,9 +1096,48 @@ async def get_room_summary(
 
                 room_summary["membership"] = membership or "leave"
         else:
-            # TODO federation API, descoped from initial unstable implementation
-            #      as MSC needs more maturing on that side.
-            raise SynapseError(400, "Federation is not currently supported.")
+            # Reuse the hierarchy query over federation
+            if remote_room_hosts is None:
+                raise SynapseError(400, "Missing via to query remote room")
+
+            fed_rooms = await self._summarize_remote_room(
+                _RoomQueueEntry(room_id, remote_room_hosts),
+                suggested_only=True,
+                max_children=1,
+                exclude_rooms=[],
+            )
+
+            for room_entry in fed_rooms:
+                room = room_entry.room
+                fed_room_id = room_entry.room_id
+
+                if fed_room_id == room_id:
+                    # If no user is specified, test against an invalid userid
+                    if requester is None:
+                        requester = ""
+
+                    # The results over federation might include rooms that the we,
+                    # as the requesting server, are allowed to see, but the requesting
+                    # user is not permitted see.
+                    #
+                    # Filter the returned results to only what is accessible to the user.
+                    if await self._is_remote_room_accessible(
+                        requester, fed_room_id, room
+                    ):
+                        # Before returning to the client, remove the allowed_room_ids
+                        # and allowed_spaces keys.
+                        room.pop("allowed_room_ids", None)
+                        room.pop("allowed_spaces", None)
+
+                        # A remote room can't be in the joined state
+                        room["membership"] = "leave"
```

Comment:
Will this be inaccurate if the user has been invited to a remote room?

Reply:
I think for invites we already store a shallow copy of the room locally, so we shouldn't end up in this case, but I am not too familiar with how that works, so actually no idea! Same about knocks, I suppose...

Reply:
> tl;dr. what does is_host_joined return for a remote room that a local user has received an invite to?

I've not read the context of this PR, I'm afraid, but `is_host_joined` looks at `current_state_events`, which will be empty for a room we are not a current member of - whether we have received an invite or not.

Reply:
Is there any function, that can return if for a room we are invited, in the knock state or joined?

Reply:
`_is_local_room_accessible` has a check for invites to rooms that we don't have state for. Maybe it can be copied?

Reply:
I guess I can just test that. However that wouldn't solve knocks, I think

Reply:
Okay, I think I found a solution (with anoa's help): `get_local_current_membership_for_user_in_room` seems to also work for remote rooms unless it is otherwise modified by purging the room for example. We tested it with knocks and invites.